### PR TITLE
[ELY-1368] Ensure the error stream is continually read so it can not cause the process to block.

### PR DESCRIPTION
**Pull request by darranl**

### CODE REVIEW

File: src/main/java/org/wildfly/security/credential/source/CommandCredentialSource.java [hunk 1]
```diff
@@ -88,10 +91,23 @@ public SupportLevel getCredentialAcquireSupport(final Class<? extends Credential
                 throw new UndeclaredThrowableException(throwable);
             }
         }
-        try {
+        try (final InputStream errorStream = process.getErrorStream()) {
```
Comment (dmlloyd) [inline src/main/java/org/wildfly/security/credential/source/CommandCredentialSource.java]: A note that as of 9, ProcessBuilder.Redirect.DISCARD can be used to safely discard output.  But that simply translates into a file redirect to "NUL" on windows and "/dev/null" on other platforms, which is regarded to be safe and also prevents the requirement to start a thread (and possible weirdness in the event that the thread creation fails).

Comment (dmlloyd) [reply src/main/java/org/wildfly/security/credential/source/CommandCredentialSource.java]: The literal code for that in OpenJDK is simply:

```
        private static final File NULL_FILE = new File(
                (GetPropertyAction.privilegedGetProperty("os.name")
                        .startsWith("Windows") ? "NUL" : "/dev/null")
        );

```

**Comment by darranl** [comment]
I can convert to redirecting to NUL and /dev/null if we are happy that is safe.

**Comment by darranl** [comment]
I am going to close this PR so we can also resubmit against 1.1.x so this is affecting PRs against that branch.

**Pull request closed without merging**